balena/models/release.py

```diff
@@ -293,3 +293,131 @@ def finalize(self, commit_or_id):
             'release', 'PATCH', params=params, data=data,
             endpoint=self.settings.get('pine_endpoint')
         )
+
+    def set_is_invalidated(self, commit_or_id, is_invalidated=False):
+        """
+        Set the is_invalidated property of a release to True or False.
+
+        Args:
+            commit_or_id (str): release commit (str) or id (int).
+            is_invalidated (bool): True for invalidated, False for validated. Default to False
+
+        Returns:
+            OK
+
+        Raises:
+            ReleaseNotFound: if release couldn't be found.
+
+        """
+
+        id = self.get(commit_or_id)['id']
+
+        params = {
+            'filter': 'id',
+            'eq': id
+        }
+        
+        data = {
+            'is_invalidated': is_invalidated
+        }
+
+        return self.base_request.request(
+            'release', 'PATCH', params=params, data=data,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+
+    def set_note(self, commit_or_id, note):
+        """
+        Set a note for a release.
+
+        Args:
+            commit_or_id (str): release commit (str) or id (int).
+            note (str): the note.
+
+        Returns:
+            OK
+
+        Raises:
+            ReleaseNotFound: if release couldn't be found.
+
+        """
+
+        id = self.get(commit_or_id)['id']
+
+        params = {
+            'filter': 'id',
+            'eq': id
+        }
+        
+        data = {
+            'note': note
+        }
+
+        return self.base_request.request(
+            'release', 'PATCH', params=params, data=data,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+
+    def set_known_issue_list(self, commit_or_id, known_issue_list):
+        """
+        Set a known issue list for a release.
+
+        Args:
+            commit_or_id (str): release commit (str) or id (int).
+            known_issue_list (str): the known issue list.
+
+        Returns:
+            OK
+
+        Raises:
+            ReleaseNotFound: if release couldn't be found.
+
+        """
+
+        id = self.get(commit_or_id)['id']
+
+        params = {
+            'filter': 'id',
+            'eq': id
+        }
+        
+        data = {
+            'known_issue_list': known_issue_list
+        }
+
+        return self.base_request.request(
+            'release', 'PATCH', params=params, data=data,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+
+    def set_release_version(self, commit_or_id, release_version):
+        """
+        Set a version for a release.
+
+        Args:
+            commit_or_id (str): release commit (str) or id (int).
+            release_version (str): the release version, only support setting the 3-digit semver values.
```

Comment:
It can actually be any semver valid value now :)
```suggestion
            release_version (str): the release version, only supports semver compliant values.
```